flow/Trace.cpp

```diff
@@ -581,12 +581,23 @@ TraceEventFields LatestEventCache::getLatestError() {
 	return latestErrors[getAddressIndex()];
 }
 
-static TraceLog g_traceLog;
+TraceLog* g_traceLog = new TraceLog("xml");
+
+bool selectTraceFormatter(std::string format) {
+	std::transform(format.begin(), format.end(), format.begin(), ::tolower);
+	if (format == "xml" || format == "json") {
+		delete g_traceLog;
+		g_traceLog = new TraceLog(format);
```

Comment:
I think this has the potential to lose log events that occur prior to selecting the formatter. To avoid that, you could instead replace the formatter inside the existing `TraceLog` and disallow (probably by ASSERT) the formatter being changed after the trace log is opened.

Reply:
Done